src/services/swap_service.ts

```diff
@@ -365,35 +366,66 @@ export class SwapService {
         const buyTokenToEthRate = makerTokenToEthRate
             .times(new BigNumber(10).pow(wethToken.decimals - makerTokenDecimals))
             .decimalPlaces(makerTokenDecimals);
-
-        const apiSwapQuote: GetSwapQuoteResponse = {
-            chainId: CHAIN_ID,
-            price,
-            guaranteedPrice,
-            to,
-            data,
-            value: adjustedValue,
-            gas: worstCaseGasEstimate,
-            estimatedGas: conservativeBestCaseGasEstimate,
-            from: takerAddress,
-            gasPrice,
-            protocolFee,
-            minimumProtocolFee: BigNumber.min(protocolFee, bestCaseProtocolFee),
-            // NOTE: Internally all ETH trades are for WETH, we just wrap/unwrap automatically
-            buyTokenAddress: isETHBuy ? ETH_TOKEN_ADDRESS : buyToken,
-            sellTokenAddress: isETHSell ? ETH_TOKEN_ADDRESS : sellToken,
-            buyAmount: makerAmount.minus(buyTokenFeeAmount),
-            sellAmount: totalTakerAmount,
-            sources: serviceUtils.convertSourceBreakdownToArray(sourceBreakdown),
-            orders: swapQuote.orders,
-            allowanceTarget,
-            decodedUniqueId,
-            sellTokenToEthRate,
-            buyTokenToEthRate,
-            quoteReport,
-            extendedQuoteReportSources,
-            priceComparisonsReport,
-        };
+        let apiSwapQuote: GetSwapQuoteResponse;
+        switch (CHAIN_ID) {
+            case CELO_CHAIN_ID:
+                apiSwapQuote = {
+                    chainId: CHAIN_ID,
+                    price,
+                    guaranteedPrice,
+                    to,
+                    data,
+                    value: adjustedValue,
+                    gas: worstCaseGasEstimate,
+                    estimatedGas: conservativeBestCaseGasEstimate,
+                    from: takerAddress,
+                    gasPrice,
+                    protocolFee,
+                    minimumProtocolFee: BigNumber.min(protocolFee, bestCaseProtocolFee),
+                    // NOTE: Internally all ETH trades are for WETH, we just wrap/unwrap automatically
+                    buyTokenAddress: isETHBuy ? '0x471ece3750da237f93b8e339c536989b8978a438' : buyToken,
```

Comment:
For a fresh reader they'd have no idea why this is duplicated with only some (2?) lines of difference. Leave a comment for future self why this is needed.

It's probably cleaner to have a helper function which takes the chain and returns the Native token address

Something like


```typescript
function nativeAssetForChain(chainId: ChainId): string => {
    switch (chainId) {
        case ChainId.Celo:
            return '0x471ece3750da237f93b8e339c536989b8978a438';
        default:
            return ETH_TOKEN_ADDRESS;
    }
}
```